lib/widgets/compose_box.dart

```diff
@@ -2021,6 +2025,21 @@ class _ComposeBoxState extends State<ComposeBox> with PerAccountStoreAwareStateM
     });
   }
 
+  @override
+  Future<void> uploadFiles(Iterable<FileToUpload> files) async {
+    await _uploadFiles(
+      context: context,
+      contentController: controller.content,
+      contentFocusNode: controller.contentFocusNode,
+      files: files);
+  }
+
+  @override
+  void insertText(String newText) {
+    final contentController = controller.content;
+    contentController.value = contentController.value.replaced(contentController.insertionIndex(), '$newText\n');
+  }
```

Comment:
How about inlining this at the call site? The caller can use the `controller` getter, which is already exposed in the ComposeBoxState interface.

(That's effectively what we do for things like quote-and-reply.)